### PR TITLE
Allow multiple HTTPS commands per request

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
@flodnv 
cc @coleaeason 

Bedrock portion of: https://github.com/Expensify/Expensify/issues/75897

This adds the ability to send more than one HTTPS request per command. This is co-requisite with: https://github.com/Expensify/Server-Expensify/pull/2610 and both must be merged together to not break anything. The correct order for this is: Merge this PR, re-run Travis on https://github.com/Expensify/Server-Expensify/pull/2610, merge https://github.com/Expensify/Server-Expensify/pull/2610 If either PR has conflicts or issues, please wait to merge both PRs.

A new (simple) test is added for this functionality. The auth tests have also been tested and verified not broken.

### CODE REVIEW

**Comment by tylerkaraszewski** [comment]
Updated, need to add tests after lunch.

File: BedrockCommand.cpp [hunk 1]
```diff
@@ -146,6 +143,16 @@ void BedrockCommand::stopTiming(TIMING_INFO type) {
     get<2>(_inProgressTiming) = 0;
 }
 
+bool BedrockCommand::httpsRequestsAreComplete() const {
```
Comment (flodnv) [inline BedrockCommand.cpp]: Our naming convention prefers `areHttpsRequestsComplete`

File: BedrockServer.h [hunk 2]
```diff
@@ -390,11 +390,23 @@ class BedrockServer : public SQLiteServer {
     // The maximum number of conflicts we'll accept before forwarding a command to the sync thread.
     atomic<int> _maxConflictRetries;
 
+    // This contains all of the command that the previous list points at. This allows us to keep only a single copy of
```
Comment (flodnv) [inline BedrockServer.h]: "previous list" ? Which previous list? :/

Comment (tylerkaraszewski) [reply BedrockServer.h]: Lol, during merge conflict resolution, I reversed the order in which `_outstandingHTTPSCommands` and `_outstandingHTTPSRequests` appeared in this file. I'll just make it explicit which list it refers to.

**Comment by tylerkaraszewski** [comment]
@flodnv updated.

File: BedrockCommand.cpp [hunk 3]
```diff
@@ -146,6 +143,16 @@ void BedrockCommand::stopTiming(TIMING_INFO type) {
     get<2>(_inProgressTiming) = 0;
 }
 
+bool BedrockCommand::AreHttpsRequestsComplete() const {
```
Comment (flodnv) [inline BedrockCommand.cpp]: Eek, sorry for the nitpick, but lowerCamelCase please 😬

Comment (tylerkaraszewski) [reply BedrockCommand.cpp]: Yes, I don't know why I did that.

**Comment by tylerkaraszewski** [comment]
@flodnv - updated again.

**Pull request merged**